### PR TITLE
Various improvements to version update checks

### LINKED ISSUE #11336: Upgrade check should only tell "... latest and greatest!" if it really is
I'm unsure if I should classify this as a bug or feature.

There has been many times in the forum (and [9819](https://discourse.metabase.com/t/why-does-metabase-want-my-smtp-password-in-plain-text/9819) and [9511](https://discourse.metabase.com/t/admin-password-reset/9511)), where people show a screenshot of Admin > Settings > Updates, where it says "You're running Metabase X.YY.Z which is the latest and greatest!", but they're not running the latest.

Errors:
1. Sometimes it's because connection to `static` is blocked.
2. Sometimes they're using their own compiled version.
3. Sometimes they have just upgraded, but the notice doesn't go away until `metabase.task.upgrade-checks.job` runs again (every 12 hours).
4. Sometimes something else is going on, which I'm not sure why.
5. Sometimes the check fails with `Sorry, we were unable to check for updates at this time.` - [not sure why](https://discourse.metabase.com/t/unable-to-check-for-updates/9893).
6. Enterprise Editions will also show `Sorry...` message

Possible changes:
1. If checks are disabled, then it should not show that message.
2. If last check was more than 1 day ago, then it should not show that message
3. If running version/hash and version-info->latest does not match, then it should recommend upgrade.
4. Have a manual button "Check for updates" together with `Last check` timestamp.
5. Host version-info over https://
6. Possibly show information from Github release notes instead of showing the short-list from version-info. Or at least link to them.
7. Link to [upgrade docs](https://www.metabase.com/docs/latest/operations-guide/upgrading-metabase.html) instead of `/start/`, when there's a newer version available. Or have an upgrade link under each type in `/start/`.
8. Enterprise Edition should either have a dedicated version-info or completely hidden
(not sure what should be done/shown, when Whitelabel is enabled - should probably not have links to metabase`.`com)
9. Hosted Cloud should show a custom message or completely hidden

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by jeff303**
# Release project changes

Updating version_info.clj code to build version-info-ee.json file for Enterprise Edition build

Adding test for both oss and ee version info file generation logic

Decoupling GitHub milestone from version and setting it to be the OSS value for :ee release build

Adding nREPL alias for release deps.edn project and usage note

# Core product changes

Adding new :ee version of :mb-version-info-url (:mb-version-info-ee-url) with default value to match the file name generated by the release code

Making enterprise-edition? public from core.clj, since it's now needed from the upgrade check code to switch between the URLs

Adding new setting to store when updates were last checked (since it's on a 12-hour interval)

Changing update check code to both set this new timestamp setting (always) to the current time, and set version-info itself to nil if the update check fails (so that users won't see "latest and greatest" if the update check succeeded at some point in the past, but then started failing

## Frontend changes

Showing the "last checked" timestamp (in user's browser timezone) when we show the "Sorry" text because of being unable to check for updates, so they know when the last check was made

Removing custom implementation of MetabaseSettings.versionInfo from EE frontend code since it's no longer needed (and never worked), because the EE backend will now be hitting a different URL with the version info specifically for EE releases

Linking to `https://www.metabase.com/docs/<newVersion>/operations-guide/upgrading-metabase.html` URL instead of simply `/start` as before

### CODE REVIEW

File: frontend/src/metabase/lib/settings.js [hunk 1]
```diff
@@ -119,6 +124,19 @@ class Settings {
...
+  versionInfoLastChecked() {
+    const ts = this.get("version-info-last-checked");
+    if (ts) {
+      // app DB stores this timestamp in UTC, so convert it to the local zone to render
+      return moment
+        .utc(parseTimestamp(ts))
+        .local()
+        .format("MMMM Do YYYY, h:mm:ss a");
+    } else {
+      return "never";
```
Comment (daltojohnso) [inline frontend/src/metabase/lib/settings.js:136]: I think this `"never"` string needs to be translated as well

File: frontend/src/metabase/lib/settings.js [hunk 2]
```diff
@@ -38,6 +41,8 @@ class Settings {
   _listeners: { [key: SettingName]: Function[] };
 
   constructor(settings: SettingsMap) {
+    console.log("settings: ");
+    console.dir(settings);
```
Comment (daltojohnso) [inline frontend/src/metabase/lib/settings.js:45]: don't forget to remove this logging

**Review by daltojohnso — CHANGES_REQUESTED** [review]

File: frontend/src/metabase/lib/settings.js (hunk 1 above)
Comment (jeff303) [reply frontend/src/metabase/lib/settings.js]: Like this?

File: frontend/src/metabase/lib/settings.js (hunk 2 above)
Comment (jeff303) [reply frontend/src/metabase/lib/settings.js]: Oops, thanks.  Removed.

**Review by daltojohnso — APPROVED** [review]
the FE looks good 👍

**Review by camsaul — APPROVED** [review]
backend stuff looks good to me as far as I can tell.

**Comment by camsaul** [comment]
I think it's safe to ignore codecov here, it seems like the existing upgrade check code doesn't have a ton of coverage but I don't want to block this PR on adding that. It's only tangentially related to the changes here

**Pull request merged**